### PR TITLE
Add nfriend's ErgoDox EZ keymap

### PR DESCRIPTION
**Pull request by nfriend**
## Description

Adds [nfriend](https://gitlab.com/nfriend)'s ErgoDox EZ keymap.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ergodox_ez/keymaps/nfriend/keymap.c [hunk 1]
```diff
@@ -0,0 +1,1095 @@
...
+/* Window management */
+#define WM_MAC 8
+#define WM_WINDOWS 9
+
+/* Navigation */
+#define ARROWS_MAC 10
+#define ARROWS_WINDOWS 11
+
+/* Misc */
+#define MISC 13
```
Comment (drashna) [inline keyboards/ergodox_ez/keymaps/nfriend/keymap.c:63]: We generally recommend using enums for layer names. As well as prepending them with underscores. 

```suggestion
enum layer_names {
    COLEMAK,
    QWERTY,
    DVORAK,
    COMMON,
    MAC,
    WINDOWS,
    NUMPAD,
    SYMBOLS,
    WM_MAC,
    WM_WINDOWS,
    ARROWS_MAC,
    ARROWS_WINDOWS,
    MISC,
};
```

File: keyboards/ergodox_ez/keymaps/nfriend/keymap.c [hunk 2]
```diff
@@ -0,0 +1,1095 @@
...
+
+enum custom_keycodes {
+    RGB_SLD = SAFE_RANGE, /* Can always be here */
+    EPRM,
```
Comment (drashna) [inline keyboards/ergodox_ez/keymaps/nfriend/keymap.c:67]: This isn't needed, as you can use `EEP_RST` for this, as it's a core code.
```suggestion
```

File: keyboards/ergodox_ez/keymaps/nfriend/rules.mk [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Set any rules.mk overrides for your specific keymap here.
+# See rules at https://docs.qmk.fm/#/config_options?id=the-rulesmk-file
+DEBOUNCE_TYPE = eager_pk
+EXTRAFLAGS += -flto
```
Comment (drashna) [inline keyboards/ergodox_ez/keymaps/nfriend/rules.mk:4]:
```suggestion
LTO_ENABLE = yes
```

File: keyboards/ergodox_ez/keymaps/nfriend/rules.mk [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+COMMAND_ENABLE = no
+UNICODE_ENABLE = false
+UNICODEMAP_ENABLE = false
```
Comment (drashna) [inline keyboards/ergodox_ez/keymaps/nfriend/rules.mk:7]:
```suggestion
UNICODE_ENABLE = no
UNICODEMAP_ENABLE = no
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by nfriend** [comment]
@drashna Thanks for all the suggestions, they are much appreciated! 😄 

I applied all of your suggestions; please let me know if there are any other changes I need to make.

Back to you 🏓

**Review by drashna — COMMENTED** [review]
Just a heads, if you only need to switch the alt/gui/control mods, bootmagic may be a better way to do that. 

https://docs.qmk.fm/#/feature_bootmagic?id=keycodes

And even if bootmagic isn't enabled, these keycodes still work.

**Review by noroadsleft — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**